examples/2d/elixir_euler_nonperiodic_p4est.jl

```diff
@@ -0,0 +1,70 @@
+
```

Comment:
About the file name: We already have a non-periodic Euler test for `p4est` (the FSP I believe). Also, for the TreeMesh the "convergence test" elixir is called `euler_source_terms`. Should we also name this one thus `euler_source_terms` or is there a conceptual difference that I don't see?

Reply:
The Euler source terms example for `TreeMesh` has a non-periodic variant too, which is just called `elixir_euler_nonperiodic.jl`. We copied this for `CurvedMesh` and appended `_curved`, and now I did the same for the `P4estMesh`.
So for `TreeMesh` and `CurvedMesh`, there are two versions of this elixir, one periodic and one non-periodic.

Reply:
OK, makes sense. Thanks for clearing this up!